### PR TITLE
[SofaPhysicsAPI] Remove unreachable code

### LINKED ISSUE #3739: [applications] unreachable code
sofa-master/applications/projects/SofaPhysicsAPI/src/SofaPhysicsAPI/SofaPhysicsSimulation.cpp line 407
Code:
   sceneFileName = filename;
        m_Simulation->init(m_RootNode.get());
        return updateOutputMeshes();

        if ( useGUI ) {  // <-- unreachable
          sofa::gui::common::GUIManager::SetScene(m_RootNode.get(),cfilename);
        }

### PR DESCRIPTION
**Pull request by alxbilger**
I removed code that was unreachable. I am not sure that it is the intended behavior. I need advices (from @epernod ?).

Fixes https://github.com/sofa-framework/sofa/issues/3739

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
Thanks @alxbilger  for this nice PR :)

File: applications/projects/SofaPhysicsAPI/src/SofaPhysicsAPI/SofaPhysicsSimulation.cpp [hunk 1]
```diff
@@ -395,32 +395,17 @@ int SofaPhysicsSimulation::load(const char* cfilename)
...
     sofa::helper::BackTrace::autodump();
 
-    //bool wasAnimated = isAnimated();
     sofa::helper::system::DataRepository.findFile(filename);
     m_RootNode = m_Simulation->load(filename.c_str());
     if (m_RootNode.get())
     {
         sceneFileName = filename;
         m_Simulation->init(m_RootNode.get());
         return updateOutputMeshes();
```
Comment (epernod) [inline applications/projects/SofaPhysicsAPI/src/SofaPhysicsAPI/SofaPhysicsSimulation.cpp]: In fact could you keep returning the result of outputMeshes to not break my version:

int result = API_SUCCESS;
if (m_RootNode.get())
{
...
   result = updateOutputMeshes();
} 
else
{
  m_RootNode = m_Simulation->createNewGraph("");
  return API_SCENE_FAILED;
}
...
return result;

**Review by epernod — CHANGES_REQUESTED** [review]

File: applications/projects/SofaPhysicsAPI/src/SofaPhysicsAPI/SofaPhysicsSimulation.cpp (hunk 1 above)
Comment (alxbilger) [reply applications/projects/SofaPhysicsAPI/src/SofaPhysicsAPI/SofaPhysicsSimulation.cpp]: done

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by epernod — APPROVED** [review]

**Pull request merged**